src/main/java/dobby/command/Dobby.java

```diff
@@ -0,0 +1,316 @@
+            }
+        } catch (IOException e) {
+            System.out.println("An error occurred while saving the tasks.");
+        }
+    }
+
+    private static String formatTaskForFile(Task task) {
+        String type = task.getClass().getSimpleName();
+        String status = task.isDone() ? "1" : "0";
+        if (task instanceof Deadline) {
+            Deadline deadline = (Deadline) task;
+            return type + "|" + status + "|" + deadline.getDescription() + "|" + deadline.getBy();
+        } else if (task instanceof Event) {
+            Event event = (Event) task;
+            return type + "|" + status + "|" + event.getDescription() + "|" + event.getFromTime() + "|" + event.getToTime();
+        } else if (task instanceof Todo) {
+            return type + "|" + status + "|" + task.getDescription();
+        } else {
+            return type + "|" + status + "|" + task.getDescription();
+        }
+    }
+
+    private static Task parseTaskFromFile(String line) {
+        String[] parts = line.split("\\|");
+        String type = parts[0];
+        boolean isDone = parts[1].equals("1");
+
+        switch (type) {
+        case "Todo":
+            Todo todo = new Todo(parts[2]);
+            if (isDone) {
+                todo.markAsDone();
+            }
+            return todo;
+        case "Deadline":
+            Deadline deadline = new Deadline(parts[2], parts[3]);
+            if (isDone) {
+                deadline.markAsDone();
+            }
+            return deadline;
+        case "Event":
+            Event event = new Event(parts[2], parts[3], parts[4]);
+            if (isDone) {
+                event.markAsDone();
+            }
+            return event;
+        default:
+            return null;
+        }
+    }
+
+    private static void handleExceptions(Exception e) {
+        printSeparator();
+        if (e instanceof MissingDescriptionException) {
+            System.out.println("    Dobby thinks master should add a description here!");
+        } else if (e instanceof IllegalInputException) {
+            System.out.println("    Dobby doesn't understand master's command!");
+        } else if (e instanceof TaskAlreadyMarkedException) {
+            System.out.println("    Dobby says master's task is already marked!");
+        } else if (e instanceof TaskAlreadyUnmarkedException) {
+            System.out.println("    Dobby says master's task is already unmarked!");
+        } else if (e instanceof EmptyListException) {
+            System.out.println("    Dobby says master's list is empty!");
+        } else if (e instanceof IndexOutOfBoundsException) {
+            System.out.println("    Dobby says master's list is full!");
+        }
+        printSeparator();
+    }
+
+    private static void processCommand (String line)
+            throws MissingDescriptionException, IllegalInputException, TaskAlreadyMarkedException,
+                TaskAlreadyUnmarkedException, EmptyListException {
```

Comment:
Indentation for wrapped lines should be 8 spaces more than the parent line (line 146)